endpoints/cookie_sync.go

```diff
@@ -152,6 +134,32 @@ func (deps *cookieSyncDeps) Endpoint(w http.ResponseWriter, r *http.Request, _ h
 	enc.Encode(csResp)
 }
 
+func parseRequest(parsedReq *cookieSyncRequest, bodyBytes []byte, usersyncIfAmbiguous bool, co *analytics.CookieSyncObject, w http.ResponseWriter) bool {
```

Comment:
Rather than returning a bool, consider returning an Error and consolidating the error handling at the call site. This will also remove 2 arguments.

Reply:
If you are fine with the new error messages, finally we could:
```
 80       parsedReq := &cookieSyncRequest{}
 81 -     if !parseRequest(parsedReq, bodyBytes, deps.gDPR.UsersyncIfAmbiguous, &co, w) {
    +     if err := parseRequest(parsedReq, bodyBytes, deps.gDPR.UsersyncIfAmbiguous); err != nil {
    +         co.Status = http.StatusBadRequest
    +         co.Errors = append(co.Errors, err)
    +         http.Error(w, co.Errors[len(co.Errors)-1].Error(), co.Status)
 82           return
 83       }
 84
 85 +-- 46 lines: if len(biddersJSON) == 0 {------------------------------------------
131       w.Header().Set("Content-Type", "application/json; charset=utf-8")
132       enc := json.NewEncoder(w)
133       enc.SetEscapeHTML(false)
134       enc.Encode(csResp)
135   }
136
137 - func parseRequest(parsedReq *cookieSyncRequest, bodyBytes []byte, usersyncIfAmbiguous bool, co *analytics.CookieSyncObject, w http.ResponseWriter) bool {
    + func parseRequest(parsedReq *cookieSyncRequest, bodyBytes []byte, usersyncIfAmbiguous bool) error {
138       if err := json.Unmarshal(bodyBytes, parsedReq); err != nil {
139 -         co.Status = http.StatusBadRequest
140 -         co.Errors = append(co.Errors, fmt.Errorf("JSON parsing failed: %s", err.Error()))
141 -         http.Error(w, co.Errors[len(co.Errors)-1].Error(), http.StatusBadRequest)
142 -         return false
    +         return fmt.Errorf("JSON parsing failed: %s", err.Error())
143       }
144
145       if parsedReq.GDPR != nil && *parsedReq.GDPR == 1 && parsedReq.Consent == "" {
146 -         co.Status = http.StatusBadRequest
147 -         co.Errors = append(co.Errors, errors.New("gdpr_consent is required if gdpr=1"))
148 -         http.Error(w, co.Errors[len(co.Errors)-1].Error(), http.StatusBadRequest)
149 -         return false
    +         return errors.New("gdpr_consent is required if gdpr=1")
150       }
151       // If GDPR is ambiguous, lets untangle it here.
152       if parsedReq.GDPR == nil {
153           var gdpr = new(int)
154           *gdpr = 1
155           if usersyncIfAmbiguous {
156               *gdpr = 0
157           }
158           parsedReq.GDPR = gdpr
159       }
160       return nil
161   }
endpoints/cookie_sync.go
```

Reply:
Would you be ok with those changes? Did I get the idea right?

Reply:
Yes. That's exactly what I was thinking. I like the unification of the error messages. The only risk is if someone had coded against the slightly different error message formats. Do you have any concern over that @hhhjort?

Reply:
If people are scripting to read the error messages, I would hope they are first checking the http status. And have something in place to flag unrecognized error messages.

Reply:
Done